File: easy/python/0101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: 'TreeNode') -> 'bool':
        if not root:
            return True
        def sym(left, right):
            if not left or not right:
                return left == right
            else:
                return left.val == right.val and sym(left.left, right.right) and sym(left.right, right.left)
        return sym(root.left, root.right)

File: easy/cpp/0101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

bool checkSym(TreeNode* p, TreeNode* q){
    if(p == NULL and q == NULL) return true;
    else if(p == NULL) return false;
    else if(q == NULL) return false;
    if(p->val != q->val) return false;
    return checkSym(p->left, q->right) && checkSym(p->right, q->left);
}

class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return checkSym(root, root);
    }
};